src/bitcoin-cli.cpp

```diff
@@ -291,6 +293,146 @@ class GetinfoRequestHandler: public BaseRequestHandler
+    const int ID_NETWORKINFO = 1;
+
+    UniValue PrepareRequest(const std::string& method, const std::vector<std::string>& args) override
+    {
+        if (!args.empty()) {
+            const std::string arg{ToLower(args.at(0))};
+            m_verbose = (arg == "true" || arg == "t" || arg == "1");
+        }
+        UniValue result(UniValue::VARR);
+        result.push_back(JSONRPCRequestObj("getpeerinfo", NullUniValue, ID_PEERINFO));
+        result.push_back(JSONRPCRequestObj("getnetworkinfo", NullUniValue, ID_NETWORKINFO));
+        return result;
+    }
+
+    UniValue ProcessReply(const UniValue& batch_in) override
+    {
+        const std::vector<UniValue> batch{JSONRPCProcessBatchReply(batch_in)};
+        if (!batch[ID_PEERINFO]["error"].isNull()) return batch[ID_PEERINFO];
+        if (!batch[ID_NETWORKINFO]["error"].isNull()) return batch[ID_NETWORKINFO];
+        // Count peer connection totals, and if m_verbose is true, store peer data in a vector of structs.
+        std::vector<m_peer> peers;
+        int ipv4_i{0}, ipv6_i{0}, onion_i{0}, block_relay_i{0}, total_i{0}; // inbound conn counters
+        int ipv4_o{0}, ipv6_o{0}, onion_o{0}, block_relay_o{0}, total_o{0}; // outbound conn counters
+        const UniValue& getpeerinfo{batch[ID_PEERINFO]["result"]};
+        for (const UniValue& peer : getpeerinfo.getValues()) {
+            const std::string addr{peer["addr"].get_str()};
+            const std::string addr_local{peer["addrlocal"].isNull() ? "" : peer["addrlocal"].get_str()};
+            const int mapped_as{peer["mapped_as"].isNull() ? 0 : peer["mapped_as"].get_int()};
+            const bool is_block_relay{!peer["relaytxes"].get_bool()};
+            const bool is_inbound{peer["inbound"].get_bool()};
+            m_conn_type conn_type{m_conn_type::ipv4};
+            if (is_inbound) {
+                if (IsAddrIPv6(addr)) {
+                    conn_type = m_conn_type::ipv6;
+                    ipv6_i += 1;
+                } else if (IsInboundOnion(mapped_as, addr, addr_local)) {
+                    conn_type = m_conn_type::onion;
+                    onion_i += 1;
+                } else {
+                    ipv4_i += 1;
+                }
+                if (is_block_relay) block_relay_i += 1;
+            } else {
+                if (IsAddrIPv6(addr)) {
+                    conn_type = m_conn_type::ipv6;
+                    ipv6_o += 1;
+                } else if (IsOutboundOnion(addr)) {
+                    conn_type = m_conn_type::onion;
+                    onion_o += 1;
+                } else {
+                    ipv4_o += 1;
+                }
+                if (is_block_relay) block_relay_o += 1;
+            }
+            if (m_verbose) {
+                const int peer_id{peer["id"].get_int()};
+                const double min_ping{peer["minping"].isNull() ? 0 : peer["minping"].get_real()};
+                const double ping{peer["pingtime"].isNull() ? 0 : peer["pingtime"].get_real()};
```

Comment:
I found ping times displayed in seconds to be somewhat uninutive. I'd personally prefer milliseconds.

(Had pings under one second to most of my peers. This might be different for other users.)

Reply:
done for min ping and ping